### PR TITLE
Add segmentation results output details to task docs

### PR DESCRIPTION
**Pull request by glenn-jocher**
## Summary
- document instance segmentation `Results` output fields, mask shape, dtype, polygons, and boxes
- document semantic segmentation `semantic_mask` output shape, class-map dtype, and lack of instance polygons
- add instance-vs-semantic comparison tables to both task pages

## Validation
- `git diff --check -- docs/en/tasks/segment.md docs/en/tasks/semantic.md`

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
📚 This PR significantly improves Ultralytics docs by clearly explaining `Results` outputs across all prediction tasks, with especially strong new coverage for semantic segmentation and `SemanticMask`.

### 📊 Key Changes
- Added `semantic_mask` to the main `Results` attribute table in the [Predict docs](../modes/predict.md), documenting it as the semantic segmentation output. 🧠
- Introduced a new **“Results by Task”** section in the Predict guide, breaking down output fields, shapes, and data types for:
  - Detect
  - Segment
  - Semantic
  - Classify
  - Pose
  - OBB 📦
- Expanded `Results` method docs so `update()` now explicitly mentions semantic masks too. ✨
- Improved `Masks` documentation by clarifying:
  - `data` is a binary `torch.uint8` tensor
  - `xy` and `xyn` return polygon lists, not raw tensors 🎭
- Added a brand-new **`SemanticMask`** documentation section with:
  - description of its purpose
  - supported properties and methods
  - a simple YOLO26 semantic inference example 🖼️
- Added **“Results Output”** sections to multiple task pages so users can quickly see what each task returns:
  - classification
  - detection
  - segmentation
  - pose
  - OBB
  - semantic 📄
- Clarified the difference between **instance segmentation** and **semantic segmentation** in both segment and semantic task docs, including side-by-side comparison tables. 🔍
- Improved code comments and examples to better explain how to access top classification predictions and segmentation outputs. ✅

### 🎯 Purpose & Impact
- Makes prediction outputs much easier to understand for both beginners and experienced users. 🙌
- Reduces confusion around which `Results` fields are available for each task, especially for semantic segmentation vs instance segmentation. 🎯
- Helps developers use YOLO26 outputs correctly by documenting expected shapes, dtypes, and helper properties more explicitly. 🛠️
- Improves discoverability of task-specific outputs without forcing users to dig through API reference pages. 🚀
- Likely lowers support burden and user mistakes when parsing model outputs in custom apps and pipelines. 📉

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help streamline review; an engineer will assist you shortly. Please review the checklist below to help ensure smooth integration 📋✨

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Good documentation improvement overall, especially the clearer side-by-side comparison and mask shape descriptions. One thing needs tightening before merge: the semantic segmentation dtype claims look underspecified/possibly inaccurate and should be made implementation-dependent unless they're explicitly guaranteed in code.

💬 Posted 1 inline comment

File: docs/en/tasks/semantic.md [hunk 1]
```diff
@@ -128,6 +128,33 @@ Use a trained YOLO26n-sem model to run predictions on images.
...
+
+YOLO semantic segmentation returns one `Results` object per image. Each result stores one dense class map for the full
+image instead of a list of object masks. Pixels with the same predicted class share the same class ID, even when they
+belong to separate objects.
+
+| Attribute | Type | Shape / Format | Description |
+|---|---|---|---|
+| `result.semantic_mask` | `SemanticMask` | One mask per image | Semantic mask container for dense per-pixel class predictions. |
+| `result.semantic_mask.data` | `torch.Tensor` | `(H, W)` | Class map where each pixel value is a predicted class ID. |
+| Class map dtype | `torch.dtype` | `uint8`, `int16`, or `int32` | Uses `torch.uint8` for up to 256 classes, then larger integer dtypes as needed. |
```
Comment (UltralyticsAssistant) [inline docs/en/tasks/semantic.md]: 💡 **MEDIUM**: The dtype guidance here is more specific than the implementation appears to guarantee. `Results` only wraps the semantic mask tensor; unless the predictor explicitly downcasts it, PyTorch class-map generation commonly ends up as `int64`, not `uint8`/`int16`/`int32` based on class count. Please avoid documenting concrete dtypes here (and in the inline example above) unless they are enforced in code, otherwise users may write dtype-dependent logic that breaks.

Comment (glenn-jocher) [reply docs/en/tasks/semantic.md]: Addressed in 43ea2a24d. The semantic inference predictor explicitly downcasts the class map in `ultralytics/models/yolo/semantic/predict.py` via `_class_map_dtype()` and `class_map.to(dtype)`, selecting `torch.uint8`, `torch.int16`, or `torch.int32` based on class count. I updated the docs to say the dtype is selected by class count and removed the older "uint8 for most datasets" wording from the inline example.

**Comment by UltralyticsAssistant** [comment]
Merged! 🎉 Thank you, @glenn-jocher, for this excellent docs improvement.

“Clarity is the soul of efficiency.” — Louis Brandeis

This PR really lives up to that idea by making `Results` outputs far easier to understand across tasks, especially with the strong new semantic segmentation and `SemanticMask` coverage. The clearer task-by-task docs, output tables, and examples will make it much easier for users to work with YOLO26 correctly and confidently.

Really valuable work that improves both the developer experience and the quality of the docs for everyone.
